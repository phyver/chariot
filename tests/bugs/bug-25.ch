data bool where True | False : bool

val if True  a b = a
  | if False a b = b

data nat where Zero : nat | Succ : nat -> nat

val le 0 _ = True
  | le _ 0 = False
  | le (m+1) (n+1) = le m n

data tree('x) where Branch : tree('x) -> 'x -> tree('x) -> tree('x)

val
  | insert_t x (Branch l y r) =
        if (le x y)
        (Branch (insert_t y r) x l)
        (Branch (insert_t x r) y l)
