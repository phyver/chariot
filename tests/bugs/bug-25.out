>> option verbose is set to 1
>> inductive type bool was succesfully defined
>> Typing for if successful
>> the definition for if is complete
>> the definition if is provably correct

>> inductive type nat was succesfully defined
>> Typing for le successful
>> the definition for le is complete
>> the definition le is provably correct

>> inductive type tree was succesfully defined
>> Typing for insert_t successful
>> the definition for insert_t is complete
>> the definition insert_t is provably correct

